packages/runtime/container-runtime/src/gc/gcConfigs.ts

```diff
@@ -162,23 +146,16 @@ export function generateGCConfigs(
 	const throwOnInactiveLoad: boolean | undefined = createParams.gcOptions.throwOnInactiveLoad;
 
 	const throwOnTombstoneLoadConfig =
-		mc.config.getBoolean(throwOnTombstoneLoadOverrideKey) ??
-		createParams.gcOptions[gcDisableThrowOnTombstoneLoadOptionName] !== true;
+		mc.config.getBoolean(throwOnTombstoneLoadOverrideKey) ?? true;
 	const throwOnTombstoneLoad =
-		throwOnTombstoneLoadConfig && sweepAllowed && !createParams.isSummarizerClient;
-	const throwOnTombstoneUsage =
-		mc.config.getBoolean(throwOnTombstoneUsageKey) === true &&
-		sweepAllowed &&
-		!createParams.isSummarizerClient;
+		throwOnTombstoneLoadConfig && sweepEnabled && !createParams.isSummarizerClient;
 
 	return {
-		gcEnabled, // For this document
-		sweepEnabled: sweepAllowed, // For this document (based on current GC Generation option)
+		gcAllowed, // For this document
+		sweepEnabled, // For this session
```

Comment:
RE: Annotations on gcAllowed and sweepEnabled
Seems like these can be confusing without annotation. Hence, suggest naming them more explicitly (as suggested in another comment I left)

Reply:
Replied to your other comment.